R/EventStudy.R

```diff
@@ -184,8 +191,15 @@ EventStudy <- function(estimator, data, outcomevar, policyvar, idvar, timevar, c
         stop("timevar column in dataset should be a vector of integers.")
     }
 
-    data_ids <- as.data.frame(data)[, c(idvar, timevar)]
+    if (is.data.table(data)) {
+        data <- data.table::copy(data)
+    } else {
+        data <- data.table::as.data.table(data)
+    }
+    data.table::setorderv(data, c(idvar, timevar))
+    data_ids <- data[, .(get(idvar), get(timevar))]
```

Comment:
I think it's better to do this as follows:
```suggestion
    data_ids <- data[, .SD, .SDcols = c(idvar, timevar)]
```